c7n/policy.py

```diff
@@ -207,14 +207,15 @@ def provision(self):
 execution = PluginRegistry('c7n.execution')
 
 
-@execution.register('pull')
-class PullMode(PolicyExecutionMode):
-    """Pull mode execution of a policy.
+class ResourceBasedMode(PolicyExecutionMode):
+    """Abstract execution mode of a simple resource-based policy.
 
-    Queries resources from cloud provider for filtering and actions.
+    Subclasses must override list_resources method (see the description).
     """
-
-    schema = utils.type_schema('pull')
+    def list_resources(self):
```

Comment:
why does this change exist? i've already mentioned this is the wrong place to hook resource retrieval. and all i see are two subclasses that copies and pastes the same implementation twice.